keras/layers/normalization.py

```diff
@@ -128,6 +132,14 @@ def build(self, input_shape):
         self.built = True
 
     def call(self, inputs, training=None):
+
+        # MXNet Backend calculates moving mean and moving variance in batch_norm operation directly.
+        # Hence, calling native mxnet batchnorm.
+        # This is functional for now, however, needs to be revisited to do it in native Keras way.
+        if K.backend() == 'mxnet':
+            return K.mxnet_batchnorm(inputs, self.gamma, self.beta, self.moving_mean, self.moving_variance,
+                                     self.momentum, axis=self.axis, epsilon=self.epsilon)
```

Comment:
I don't understand how this can possibly work. BN has a different behavior in inference and training, but it doesn't appear to be reflected here.

What is the problem with using just the backend ops to implement BN?

Reply:
In the backend, for executing the graph with MXNet module (Executor), a flag is set to indicate whether the stage is training or inference. That was managing different behavior required.